Tella/Scenes/Settings/ViewModel/GDriveServerViewModel.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  GDriveServerViewModel.swift
+//  Tella
+//
+//  Created by gus valbuena on 5/22/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class GDriveServerViewModel: ObservableObject {
+    var mainAppModel : MainAppModel
+    private let gDriveRepository: GDriveRepositoryProtocol
+    private var cancellables = Set<AnyCancellable>()
+
+    @Published var selectedDrive: SharedDrive? = nil
+    @Published var sharedDriveState: ViewModelState<[SharedDrive]> = .loading
+    @Published var createFolderState: ViewModelState<String> = .loaded("")
+    
+    init(repository: GDriveRepositoryProtocol, mainAppModel: MainAppModel) {
+        self.mainAppModel = mainAppModel
+        self.gDriveRepository = repository
+    }
+
+    func getSharedDrives() {
+        gDriveRepository.getSharedDrives()
+            .receive(on: DispatchQueue.main)
+            .sink(receiveCompletion: {completion in
+                switch completion {
+                case .finished:
+                    break
+                case.failure(let error):
+                    self.sharedDriveState = .error(error.localizedDescription)
```

Comment:
It might be better to standardize the return error type as `APIError` and use `.error(error.errorMessage)`

Reply:
got it, I just pushed it. 

In my opinion, the refactor of the APIError class is a bit out of scope for this PR. It’s something we could have easily merged after this one. Let’s try to avoid adding more refactors and code changes to large epic PRs because they keep getting bigger, making the release time longer and further delays in other features

Reply:
but if you agree handleSignIn(), getSharedDrives() and createDriveFolder() are in gDrive feature are not related  to refactoring
and In my opinion I don't think adding custom gDrive error messages are APIError refactoring